text_localization_environment/TextLocEnv.py

```diff
@@ -280,7 +283,10 @@ def get_warped_bbox_contents(self):
         return cropped.resize((224, 224), LANCZOS)
 
     def compute_state(self):
-        return np.concatenate((self.extract_features().array, np.array(self.history).flatten()))
+        if self.gpu_id != -1:
+            return np.concatenate((cuda.to_cpu(self.extract_features().array), np.array(self.history).flatten()))
+        else:
+            return np.concatenate((self.extract_features().array, np.array(self.history).flatten()))
```

Comment:
For reduced code duplication and increased readability, I'd suggest refactoring the code to something like the following

```python
features = cuda.to_cpu(self.extract_features().array) if self.gpu_id != -1 else self.extract_features().array
history = np.array(self.history).flatten()
return np.concatenate(features, history)
```